Lib/test/test_fractions.py

```diff
@@ -355,6 +355,29 @@ def testInitFromDecimal(self):
         self.assertRaises(OverflowError, F, Decimal('inf'))
         self.assertRaises(OverflowError, F, Decimal('-inf'))
 
+    def testInitFromIntegerRatio(self):
+        class Ratio:
+            def __init__(self, ratio):
+                self._ratio = ratio
+            def as_integer_ratio(self):
+                return self._ratio
+        class RatioNumber(Ratio):
+            pass
+        numbers.Number.register(RatioNumber)
+
+        self.assertEqual((7, 3), _components(F(RatioNumber((7, 3)))))
```

Comment:
```suggestion
        # as_integer_ratio() output should be normalized; lets check that
        # we don't touch this
        self.assertEqual((6, 4), _components(F(RatioNumber((6, 4)))))
```

Forgot that.  I think, this should address Victor's note.

Reply:
This is invalid ratio. The result is undefined.

Reply:
> This is invalid ratio.

Yeah, but test just checks that we don't touch as_integer_ratio() output.